source/MaterialXTest/ShaderGen.cpp

```diff
@@ -225,6 +225,13 @@ void createLightRig(mx::DocumentPtr doc, mx::HwLightHandler& lightHandler, mx::H
     lightHandler.bindLightShaders(shadergen);
 }
 
+static std::string RESULT_DIRECTORY("results/");
```

Comment:
This was the only way I could figure out to make this be called first to create the "results' directory.
I looked at your makedirectory() function and it is allowed to failed if the directory already exists.